src/lib/containers/UserCard.tsx

```diff
@@ -1,108 +1,95 @@
   disableLink?: boolean
-  extraSmall?: boolean
   isCertified?: boolean
   isValidated?: boolean
 }
 
-export default class UserCard extends React.Component<
-  UserCardProps,
-  UserCardState
-> {
-  constructor(props: UserCardProps) {
-    super(props)
-    this.state = { userProfile: undefined, isLoading: true, preSignedURL: '' }
-    this.getUserProfile = this.getUserProfile.bind(this)
-  }
+export const UserCard: React.FunctionComponent<UserCardProps> = (
+  props: UserCardProps,
+) => {
+  const {
+    userProfile: initialProfile,
+    preSignedURL: initialPreSignedURL,
+    size,
+    ownerId,
+    alias,
+    token,
+    ...rest
+  } = props
+  const [userProfile, setUserProfile] = useState(initialProfile)
+  const [principalId, setPrincipalId] = useState(ownerId)
+  const [isLoading, setIsLoading] = useState(true)
+  const [preSignedURL, setPresignedUrl] = useState(initialPreSignedURL ?? '')
 
-  public componentDidMount() {
-    const { userProfile, ownerId, alias, token } = this.props
+  useEffect(() => {
     if (userProfile) {
-      return
-    }
-    if (alias) {
+      setIsLoading(false)
+    } else if (alias) {
+      // Before we can get the profile, we must get the principal ID using the alias
       getPrincipalAliasRequest(token, alias, 'USER_NAME').then(
         (aliasData: any) => {
-          this.getUserProfile(aliasData.principalId!)
+          setPrincipalId(aliasData.principalId)
         },
       )
-    } else {
-      // check for ownerId!
-      this.getUserProfile(ownerId!)
     }
-  }
+  }, [userProfile, alias, token])
 
-  public getUserProfile(ownerId: string) {
-    getUserProfileWithProfilePic(ownerId!, this.props.token)
-      .then((data) => {
-        const { userProfile, preSignedURL } = data
-        this.setState({ userProfile, preSignedURL, isLoading: false })
-      })
-      .catch((err) => {
-        console.log('failed to get user bundle ', err)
-      })
-  }
-
-  public render() {
-    const {
-      userProfile,
-      preSignedURL,
-      size,
-      ...rest
-    } = this.props
-    let userProfileAtRender
-    let preSignedURLAtRender
-    if (!userProfile) {
-      // userProfile wasn't passed in from props
-      if (this.state.isLoading) {
-        // still making the API call
-        return <></>
-      }
-      userProfileAtRender = this.state.userProfile
-      preSignedURLAtRender = this.state.preSignedURL
-    } else {
-      // otherwise we have the profile from props
-      userProfileAtRender = userProfile
-      preSignedURLAtRender = preSignedURL
-    }
-    const propsForChild = {
-      userProfile: userProfileAtRender!,
-      preSignedURL: preSignedURLAtRender,
-      ...rest,
+  useEffect(() => {
+    if (!userProfile && principalId) {
+      getUserProfileWithProfilePic(principalId, token)
+        .then(data => {
+          setUserProfile(data.userProfile)
+          setPresignedUrl(data.preSignedURL)
```

Comment:
We get the presigned URL of the image up front.  If this is a small card and we wait to hover, is this presigned URL still valid after the expire time (or does it fetch it up front, before the hover card is shown)?

Reply:
Currently, it actually re-fetches the profile (including a new pre-signed URL) on hover, since I only pass the principal ID along, but I prefer your solution

Reply:
After changes, the image is fetched when the user card is first loaded